main.go

```diff
@@ -64,24 +65,46 @@ func InfoHandler(contentAPI, needAPI, performanceAPI string, config *Config) fun
 			return
 		}
 
-		needStart := time.Now()
-		for _, needID := range artefact.Details.NeedIDs {
-			need, err := need_api.FetchNeed(needAPI, config.BearerTokenNeedAPI, needID)
-			if err != nil {
-				renderError(w, http.StatusInternalServerError, "Need: "+err.Error())
+		var waitGroup sync.WaitGroup
+		errorChannel := make(chan string)
+
+		waitGroup.Add(1)
+		go func() {
+			defer waitGroup.Done()
+
+			needStart := time.Now()
+			for _, needID := range artefact.Details.NeedIDs {
+				need, nErr := need_api.FetchNeed(needAPI, config.BearerTokenNeedAPI, needID)
+				if nErr != nil {
+					errorChannel <- "Need:" + nErr.Error()
```

Comment:
Sorry, I missed this on the previous pass:

This code will currently deadlock on error. This is writing to an unbuffered channel, and will therefore block until there's another goroutine ready to read from the channel. The reads in this code don't happen until after the `waitGroup.Wait()` call below, which won't return until both of these goroutines complete.

The simplest fix is probably to make `errorChannel` a buffered channel with enough capacity to hold all the errors it could potentially receive.